Providers/Profile.swift

```diff
@@ -60,40 +60,76 @@ class CommandDiscardingSyncDelegate: SyncDelegate {
  * This will also likely be the extension point for wipes, resets, and
  * getting access to data sources during a sync.
  */
+
+let TabSendURLKey = "TabSendURL"
+let TabSendTitleKey = "TabSendTitle"
+let TabSendCategory = "TabSendCategory"
+
+enum SentTabAction: String {
+    case View = "TabSendViewAction"
+    case Bookmark = "TabSendBookmarkAction"
+    case ReadingList = "TabSendReadingListAction"
+}
+
 class BrowserProfileSyncDelegate: SyncDelegate {
     let app: UIApplication
 
     init(app: UIApplication) {
         self.app = app
+        registerForNotifications()
     }
 
     // SyncDelegate
     func displaySentTabForURL(URL: NSURL, title: String) {
         log.info("Displaying notification for URL \(URL.absoluteString)")
 
-        app.registerUserNotificationSettings(UIUserNotificationSettings(forTypes: UIUserNotificationType.Alert, categories: nil))
-        app.registerForRemoteNotifications()
+        let currentSettings = app.currentUserNotificationSettings()
+        if currentSettings.types.rawValue & UIUserNotificationType.Alert.rawValue != 0 {
+            // TODO: localize.
+            let notification = UILocalNotification()
+
+            notification.fireDate = NSDate()
+            notification.timeZone = NSTimeZone.defaultTimeZone()
+            notification.alertBody = String(format: NSLocalizedString("New tab: %@: %@", comment:"New tab [title] [url]"), title, URL.absoluteString!)
+            notification.userInfo = [TabSendURLKey: URL.absoluteString!, TabSendTitleKey: title]
+            notification.alertAction = nil
+            notification.category = TabSendCategory
+
+            app.presentLocalNotificationNow(notification)
+        }
+    }
+
+
+    func registerForNotifications() {
+        let viewAction = UIMutableUserNotificationAction()
+        viewAction.identifier = SentTabAction.View.rawValue
+        viewAction.title = NSLocalizedString("View", comment: "View a URL")
```

Comment:
The space for these buttons is pretty tight which may not work out well for some languages. You should probably mention that in the comment. Sometimes I upload a screenshot to Bugzilla or DropBox and put a link to that in the comment so that the localizers have a good idea what it will look like.